docker/Dockerfile

```diff
@@ -8,88 +8,38 @@
 # Install SCIP solver (from source)
 
-ARG SCIP_VERSION=7.0.3
+ARG SCIP_VERSION=8.0.3
 ARG IPOPT_VERSION=3.14.4
 
-FROM ubuntu:latest as COMPILE
-
-ARG SCIP_VERSION
-ARG IPOPT_VERSION
-
-ENV DEBIAN_FRONTEND=noninteractive
-
-RUN apt-get update \
-  && apt-get install -y --no-install-recommends g++ gfortran make cmake ca-certificates wget   \
-  pkg-config libblas3 libgfortran4 libgsl23 libgslcblas0 liblapack3 libtbb2 libboost-program-options1.71 \
-    build-essential libgmp-dev libblas-dev libbliss-dev zlib1g-dev libgsl-dev libcliquer-dev \
-    libreadline-dev liblapack-dev git file \
-  && rm -rf /var/lib/apt/lists/*
-
-WORKDIR /opt
-
-# Install hsl (IpOpt dependency)
-COPY ./docker/coinhsl-2021.05.05.tar.gz $WORKDIR
-RUN git clone https://github.com/coin-or-tools/ThirdParty-HSL.git \
-  && cd ThirdParty-HSL \
-  && tar xzvf ../coinhsl-2021.05.05.tar.gz \
-  && mv coinhsl-2021.05.05 coinhsl \
-  && ./configure \
-  && make \
-  && make install
-
-# Install IpOpt from source (SCIP dependency)
-RUN cd $WORKDIR
-RUN wget https://github.com/coin-or/Ipopt/archive/refs/tags/releases/${IPOPT_VERSION}.tar.gz \
-  && tar xzvf ${IPOPT_VERSION}.tar.gz \
-  && cd Ipopt-releases-${IPOPT_VERSION} \
-  && mkdir build \
-  && cd build \
-  && ADD_CFLAGS=-fopenmp ADD_FFLAGS=-fopenmp ADD_CXXFLAGS=-fopenmp ../configure --with-hsl-cflags="-I/usr/local/include/coin-or/hsl/" --with-hsl-lflags="-L/usr/local/lib/ -lcoinhsl" \
-  && make \
-  && make install
-
-RUN cd $WORKDIR
-RUN wget https://scip.zib.de/download/release/scipoptsuite-${SCIP_VERSION}.tgz --no-check-certificate  \
-  && tar xzf scipoptsuite-${SCIP_VERSION}.tgz \
-  && cd scipoptsuite-${SCIP_VERSION} \
-  && mkdir build \
-  && cd build \
-  && cmake -DZIMPL=off .. \
-  && make CXX_FLAGS="-DSOPLEX_WITH_GMP -fPIC" install \
-  && cd $WORKDIR
-
-# Runtime image
-
-FROM ubuntu:latest
+FROM ubuntu:20.04
 
 ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update \
   && apt-get install -y --no-install-recommends \
     wget libblas3 libgfortran4 libgsl23 libgslcblas0 liblapack3 libtbb2 libboost-program-options1.71 lp-solve \
     build-essential cmake g++-9 libgmp-dev libblas-dev libbliss-dev zlib1g-dev libgsl-dev libcliquer-dev \
-    libreadline-dev git ca-certificates ssh liblpsolve55-dev libgomp1 \
+    libreadline-dev git ca-certificates ssh liblpsolve55-dev libgomp1 curl patchelf \
   && rm -rf /var/lib/apt/lists/*
 
 WORKDIR /opt
 
-COPY --from=COMPILE /usr/local/lib/ /usr/local/lib/
-COPY --from=COMPILE /usr/local/include/ /usr/local/include/
-
-# Install SCIP solver (binaries)
-#RUN wget https://scip.zib.de/download/release/SCIPOptSuite-7.0.2-Linux-ubuntu.sh --no-check-certificate && \
-#  chmod +x ./SCIPOptSuite-7.0.2-Linux-ubuntu.sh && \
-# ./SCIPOptSuite-7.0.2-Linux-ubuntu.sh --skip-license
+ # Install SCIP solver (binaries)
+RUN curl -L https://scip.zib.de/download/release/SCIPOptSuite-8.0.3-Linux-ubuntu.sh --output installer.sh && \
+  chmod +x ./installer.sh && \
```

Comment:
The comments seem to suggest that we need to build IpOpt for "faster HSL" code. In fact, it looks like the installed IpOpt version is [3.13.3](https://scipopt.org/index.php#download) while we were previously building 3.14.4.

Not sure how important this is, but wanted to point it out.

Reply:
This dependency is only used for building `SCIP`. Since this gets distributed with their pre built library I think it should be fine. Very perceptive, though. 👍